src/ziggurat/messaging/producer.clj

```diff
@@ -43,33 +44,40 @@
        (sentry/report-error sentry-reporter e "Error while declaring RabbitMQ queues")
        (throw e)))))
 
+(defn- record-headers->map [record-headers]
+  (reduce (fn [header-map record-header]
+            (assoc header-map (.key record-header) (String. (.value record-header)))),
+          {},
+          record-headers))
+
 (defn- properties-for-publish
-  [expiration]
+  [expiration headers]
   (let [props {:content-type "application/octet-stream"
-               :persistent   true}]
+               :persistent   true
+               :headers      (record-headers->map headers)}]
```

Comment:
where are these headers being used? do you think it makes sense to add them to the message-payload?

Reply:
rabbit expects the headers this way as properties